FeedStoreChallenge/CoreDataFeedStore.swift

```diff
@@ -29,14 +29,51 @@ public final class CoreDataFeedStore: FeedStore {
 	}
 
 	public func retrieve(completion: @escaping RetrievalCompletion) {
-		fatalError("Must be implemented")
+		let context = self.context
+		context.perform {
+			do {
+				let request = NSFetchRequest<ManagedCache>(entityName: "\(ManagedCache.self)")
+				request.returnsObjectsAsFaults = false
+				if let cache = try context.fetch(request).first {
+					completion(.found(feed: cache.feed.compactMap { ($0 as? ManagedFeedImage) }.map { LocalFeedImage(id: $0.id, description: $0.imageDescription, location: $0.location, url: $0.url) }, timestamp: cache.timestamp))
```

Comment:
All other methods in `CoreDataFeedStore` are using helpers to make code more readable and concise. In this line `ManagedFeedImage`s are mapped to `LocalFeedImage`s inline. The code would be shorter and easier to read if this transformation is moved to a separate helper.

Please move mapping to a separate helper.

```suggestion
					completion(.found(feed: cache.localFeedImages(), timestamp: cache.timestamp))
```